src/main/java/coydir/logic/commands/RateCommand.java

```diff
@@ -59,6 +61,12 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
 
+        for (Rating ratinglist : targetPerson.getRatingHistory()) {
+            if (ratinglist.getTime().equals(rating.getTime())) {
+                throw new CommandException(MESSAGE_EMPLOYEE_RATING_COMPLETED);
```

Comment:
Maybe something to think about (don't need to change). Given that you can only add one rating a day, your last rating in the list would be the only date that you might check against. Then you would not need a for loop. But since there is no time sensitivity, not super necessary